pkg/api/validate/format_regexps.go

```diff
@@ -15,4 +15,6 @@ var (
 		`([a-z0-9]|[a-z0-9][-a-z0-9]{0,61}[a-z0-9])` +
 		`(\.([a-z0-9]|[a-z0-9][-a-z0-9]{0,61}[a-z0-9]))*` +
 		`$`)
+	// https://tools.ietf.org/html/rfc1035
+	RxRFC1035 = regexp.MustCompile(`^[a-z]([-a-z0-9]*[a-z0-9])?$`)
```

Comment:
I'm -1.
a) we already validate this in pkg/frontend/middleware/validate.go, so this is the wrong place.
b) it adds further restrictions post-GA, which could be problematic with existing clusters.
c) RFC1035 is a combination of regexp and length check, the latter is omitted here.
d) This doesn't help #671 because that's a domain issue not a cluster name issue.